jscomp/others/js_dict.mli

```diff
@@ -78,6 +78,6 @@ val fromArray : (key * 'a) array -> 'a t
 (** [fromArray entries] creates a new dictionary using with containing each
 [(key, value)] pair in [entries] *)
 
-val map : ('a -> 'b) -> 'a t -> 'b t
+val map : ('a -> 'b [@bs]) -> 'a t -> 'b t
```

Comment:
Yes, uncurried calling convention is supported, maybe reason could provide better syntax, you can always wrap a curried function in the user land, but we only export uncurried for efficiency?